src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -146,7 +146,7 @@ private function autowireCalls(\ReflectionClass $reflectionClass, bool $isRoot):
         $this->decoratedClass = null;
         $this->getPreviousValue = null;
 
-        if ($isRoot && ($definition = $this->container->getDefinition($this->currentId)) && $this->container->has($this->decoratedId = $definition->innerServiceId)) {
+        if ($isRoot && ($definition = $this->container->getDefinition($this->currentId)) && null !== ($this->decoratedId = $definition->innerServiceId) && $this->container->has($this->decoratedId)) {
```

Comment:
Given that the interface were always documented as accepting string, maybe this should be applied in older branches, so that we don't violate our own contract.

Reply:
I'll do that.

Reply:
#40386